### PR TITLE
Visualize plan, obstacles and via points also on failure

### PR DESCRIPTION
**Pull request by corot**
Particularly useful to see diverged trajectories, e.g.

https://user-images.githubusercontent.com/322610/230597737-f16e68c5-4e4c-46b8-8e5e-4b3964427e8c.mp4

Helped on debugging https://www.wrike.com/open.htm?id=1091406204

### CODE REVIEW

**Review by siferati — APPROVED** [review]
Can you show a video of what it looks like when the planning fails?
I get the point, but isn't it confusing to be seeing a trajectory in rviz that actually failed?
I think it will lead us to think "why is the robot not moving even though it has a plan??"
None of the other planners behave like this

File: src/teb_local_planner_ros.cpp [hunk 1]
```diff
@@ -358,11 +358,18 @@ uint32_t TebLocalPlannerROS::computeVelocityCommands(const geometry_msgs::PoseSt
   // Now perform the actual planning
 //   bool success = planner_->plan(robot_pose_, robot_goal_, robot_vel_, cfg_.goal_tolerance.free_goal_vel); // straight line init
   bool success = planner_->plan(transformed_plan, &robot_vel_, cfg_.goal_tolerance.free_goal_vel);
+
+  // Visualize plan, obstacles and via points
+  planner_->visualize();
```
Comment (renan028) [inline src/teb_local_planner_ros.cpp:363]: Q: is it possible to add the failed trajectories to other namespace? (e.g `failed_plan`)

Comment (corot) [reply src/teb_local_planner_ros.cpp:363]: must be other topic;  trajectories are shown as a pose array, no visual markers

Comment (corot) [reply src/teb_local_planner_ros.cpp:363]: would be nice to show them with a different color... :thinking: 
will check if I can do fast

Comment (corot) [reply src/teb_local_planner_ros.cpp:363]: kinda,,, but it's gonna be rather ugly, so not worth it

**Review by renan028 — APPROVED** [review]

**Pull request merged**